packages/block-library/src/calendar/index.php

```diff
@@ -115,7 +115,7 @@ function block_core_calendar_update_has_published_posts() {
 	 *
 	 * @param int $post_id Deleted post ID.
 	 */
-	function block_core_calendar_update_has_published_post_on_delete( $post_id ) {
+	function gutenberg_block_core_calendar_update_has_published_post_on_delete( $post_id ) {
```

Comment:
This can be a problem. I think block PHP files are automatically copied into WP core during package updates, and we can't have the `gutenberg_` prefix there.

Reply:
:thinking: This and the other change in this file were causing fatals as well. What would be the proper fix?

Reply:
Sorry, I'm not sure. I know Nik (@ntsekouras) renamed a bunch of functions last week, maybe he knows what's the proper fix.

Reply:
> I think block PHP files are automatically copied into WP core during package updates, and we can't have the gutenberg_ prefix there.

That's correct. How this function creates any problem as is only used from this block? It would be a problem if it did call a `gutenberg` prefixed function that made it into core ([example PR](https://github.com/WordPress/gutenberg/pull/36180/).

Reply:
Hey, this is what I know. Without these fixes I get this error when activating the plugin (11.8.2 version) with the latest WordPress core:

![Captura de ecrã de 2021-11-09 11-38-45](https://user-images.githubusercontent.com/583546/140909298-79c20c3e-a400-412a-85a8-5b3cd92eb47c.png)

I also see that the functions `block_core_calendar_update_has_published_post_on_transition_post_status` and `block_core_calendar_update_has_published_post_on_delete` were backported to WordPress core as of [this commit](https://github.com/WordPress/wordpress-develop/commit/fdb2dacc199382cd0800838596b5717cbf8e211f) ([see related PR](https://github.com/WordPress/wordpress-develop/pull/1827)).

So, the fact that the plugin uses the same names is why the error happens, and renaming them fixes it. What I'm unfamiliar with is how `packages/block-library/src/calendar/index.php` in Gutenberg is converted into the `src/wp-includes/blocks/calendar.php#L151` in WordPress core, but I presume is part of the package update.

Reply:
I'm changing the block calendar code to cover against the function being already present at https://github.com/WordPress/gutenberg/pull/36347/commits/601a01250e0111527a88c68cfca0caec719f85ac so it doesn't require a package update. Props @jorgefilipecosta

Reply:
Removed the conditional functions to avoid issues with the build step of Gutenberg that renames the functions in block files.